### PR TITLE
solution hw-hb-04

### PR DESCRIPTION
**Pull request by RomanHreits**

### CODE REVIEW

**Review by DmGrt — APPROVED** [review]
It looks like everything is fine

File: src/main/java/com/cinema/model/CinemaHall.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+@ToString(of = {"id", "capacity", "description"})
```
Comment (aleksandr-hrankin) [inline src/main/java/com/cinema/model/CinemaHall.java]: Посмотри что делает @Data в lombok)

**Review by aleksandr-hrankin — APPROVED** [review]
Good Job!

File: src/main/java/com/cinema/model/MovieSession.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+@EqualsAndHashCode(of = {"id", "movie", "cinemaHall", "showTime"})
+@Entity
+@Table(name = "movie_session")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Movie movie;
+    @ManyToOne(fetch = FetchType.LAZY)
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/MovieSession.java:24]: nice👍

File: src/main/java/com/cinema/model/MovieSession.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@ToString(of = {"id", "movie", "cinemaHall", "showTime"})
+@EqualsAndHashCode(of = {"id", "movie", "cinemaHall", "showTime"})
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/MovieSession.java]: google about Data annotation, that wil make list of annotation shorter

File: src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        try {
+            transaction = hibernateSession.beginTransaction();
+            hibernateSession.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add MovieSession to DB", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java]: let's make message more informative and add some info about entity we were trying to save same with other methods, where applicable

File: src/main/java/com/cinema/model/MovieSession.java (hunk 3 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/model/MovieSession.java]: done

File: src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java (hunk 4 above)
Comment (RomanHreits) [reply src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**